ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh

```diff
@@ -1,21 +1,64 @@
 #!/usr/bin/env bash
 
 set -x
+set +e
+
+BASE_DIR=${HOME:-"/alabama"}
+MUST_GATHER_TIMEOUT=${MUST_GATHER_TIMEOUT:-"5m"}
+
+function create_out_dir {
+  local OUT_DIR=${1}
+  if [ ! -d "./${OUT_DIR}" ]; then
+    mkdir ${OUT_DIR}
+  fi
+}
+
+function gen_pods_debug_cmd {
+  create_out_dir ${1}
+  local cmd=("echo 'Logs for pod: {}'; oc describe pod {} > ${1}/{}-describe.log; oc logs --all-containers=true {} >> ${1}/{}-containers.log")
+  echo "${cmd[@]}"
+}
+
+function gen_api_resources_debug_cmd {
+  create_out_dir ${1}
+  local cmd=("echo 'Info for api-resource: {}'; oc get {} -o yaml > ${1}/{}.log")
+  echo "${cmd[@]}"
+}
 
 # We don't want to use OpenShift-CI build cluster namespace
 unset NAMESPACE
-
 oc project openstack
 
-oc logs ceph
+pushd "${BASE_DIR}" || exit
+mkdir logs
+pushd logs || exit
 
-oc get OpenStackControlPlane openstack -o json
+oc get all > get_all.log
 
-oc get all
+CMD=$(gen_api_resources_debug_cmd "resources")
+oc api-resources | grep 'openstack' | awk '{print $1}' | xargs -n1 -I {} sh -c "${CMD[@]}"
+# RabbitMQCluster
+oc get -o yaml RabbitMQCluster > RabbitMQCluster.log
 
-oc get -o yaml MariaDB,RabbitMQCluster,KeystoneAPI,PlacementAPI,Glance,Cinder,NeutronAPI,Nova
+CMD=$(gen_pods_debug_cmd "controller-manager")
+oc get pods -n openstack --show-labels | grep -i '.*control\-plane\=.*controller\-manager.*' | awk '{print $1}' | \
+xargs -n1 -I {} sh -c "${CMD[@]}"
 
-oc get pods -n openstack --show-labels | grep -i '.*control\-plane\=.*controller\-manager.*' | awk '{print $1}' | xargs -n1 oc logs
+CMD=$(gen_pods_debug_cmd "services")
+oc get pods -n openstack --no-headers | egrep -iv controller | awk '{print $1}' | xargs -n1 -I {} sh -c "${CMD[@]}"
 
+CMD=$(gen_pods_debug_cmd "errors")
 # Capture logs from service containers if container is not in Running or Completed state
-oc get pods -n openstack --no-headers | egrep -iv controller | egrep -iv 'Running|Completed' | awk '{print $1}' | xargs -n1 -I {} sh -c 'echo "Logs for pod: {}"; oc describe pod {}; oc logs --all-containers=true {}'
+oc get pods -n openstack --no-headers | egrep -iv controller | egrep -iv 'Running|Completed' | awk '{print $1}' | \
+xargs -n1 -I {} sh -c "${CMD[@]}"
+
+# must-gather
+mkdir -p ${ARTIFACT_DIR}/must-gather/
+oc --insecure-skip-tls-verify adm must-gather --timeout=$MUST_GATHER_TIMEOUT \
```

Comment:
So do we want to continue with must-gather? We can add a boolean to skip this, so we can have this code ready to run when dev need it.

Reply:
I think it still makes sense to capture must gather to check openshift logs - its possible that openshift itself can have issues, so must gather will allow us to debug issues in openshift.

Reply:
@stuggi agree with you. Since we still don't have this image, make sense we use the default one here for now, and replace with ours once is ready?